lib/elixir/lib/application.ex

```diff
@@ -4,32 +4,89 @@ defmodule Application do
 
   In Elixir (actually, in Erlang/OTP), an application is a component
   implementing some specific functionality, that can be started and stopped
-  as a unit, and which can be re-used in other systems.
+  as a unit, and which can be reused in other systems.
 
-  Applications are defined with an application file named `APP.app` where
-  `APP` is the application name, usually in `underscore_case`. The application
-  file must reside in the same `ebin` directory as the compiled modules of the
-  application. In Elixir, the Mix build tool is responsible for compiling your
-  source code and generating your application `.app` file. You can learn more
-  about the generation of `.app` files by typing `mix help compile.app`.
+  ## The application resource file
+
+  Applications are specified in their [*resource
+  file*](http://erlang.org/doc/man/app.html), which is a file called `APP.app`,
+  where `APP` is the application name. For example, the application resource
+  file of the OTP application `ex_unit` is called `ex_unit.app`.
+
+  You'll find the resource file of an application in its `ebin` directory, it is
+  generated automatically by Mix. Some of its keys are taken from the keyword
+  lists returned by the `project/0` and `application/0` functions defined in
+  `mix.exs`, and others are generated by Mix itself.
+
+  You can learn more about the generation of application resource files in the
+  [`mix compile.app`](../mix/Mix.Tasks.Compile.App.html) task.
```

Comment:
Im wondering how is this going to look like when read in IEx.
This is something I would like to propose/implement in exdoc. The ability.to link to other Elixir proyects

Reply:
I think we should say:

> files in the `Mix.Tasks.Compile.App` task (or also by running `mix help compile.app`).

Reply:
Do you prefer one of the two options?

Linking to `Mix.Tasks.Compile.App`, even if manually, is normal: you link to a module's documentation.

On the other hand, I wonder if mentioning the module is natural or not for an end-user, compared to what they normally interact with in the command-line. (I wonder in the sense that I do not know the answer.) In that option, linking `mix help compile.app`... is a bit strange because you are linking a command to the source code of a module. Not a big mismatch, but just someting to take into account.

What do you think?

(BTW, I need to work in the general structure of the module doc and merge some sections, the patch as is needs more work but all this feedback is being incorporated to the second pass.)

Reply:
If we use `Mix.Tasks.Compile.App`, then we will get automatic linking in places like ExDoc, and pointing out to `mix help compile.app` in parens will be helpful for anybody in the command line.

Given that both are public and the relationship between them is public, I think going with both is fine and the most user friendly. Does it make sense?

Reply:
Sounds good, @whatyouhide good for you too?

BTW, locally I do not get autolinking from Elixir to Mix, the purpose of the relative path is to link manually in a way that works both in edge and stable docs. Am I missing something?

Reply:
If autolinking to mix modules is not working then it should be treated as an ex_doc bug. We should investigate. :)

Reply:
Its a missing feature. As far as i know it has never been implemented

Reply:
Could work on that too if you will. In that case, could we perhaps leave here just fixed-width font by now?